app/models/miq_group.rb

```diff
@@ -69,7 +69,32 @@ def self.with_roles_excluding(identifier)
   end
 
   def self.next_sequence
-    maximum(:sequence).to_i + 1
+    # The +(current_scope || self)+ here is a result of a Rails 6.0 deprecation
+    # warning that is added here:
+    #
+    #   https://github.com/rails/rails/pull/35280
+    #
+    # This was an attempt to fix "leaking scopes", however, in our case, we use
+    # this method both for our +default_value_for(:sequence)+, and it will get
+    # used as part of +.create_tenant_group+.
+    #
+    # As such, we need both +.current_scope+ for when we want to scope down the
+    # +.next_sequence+, but also well allow it to be used on a raw +.create+
+    # without scopes.
+    #
+    # Our +.current_scope+ use case can be best described via one our of
+    # +MiqGroup+ specs:
+    #
+    #   expect(MiqGroup.next_sequence).to be < 999 # sanity check
+    #
+    #   FactoryBot.create(:miq_group, :description => "want 1", :sequence => 999)
+    #   FactoryBot.create(:miq_group, :description => "want 2", :sequence => 1000)
+    #   FactoryBot.create(:miq_group, :description => "dont want", :sequence => 1009)
+    #
+    #   expect(MiqGroup.where("description like 'want%'").next_sequence).to eq(1001)
+    #
+    #
+    (current_scope || self).maximum(:sequence).to_i + 1
```

Comment:
I'm kind of surprised we have sequences at all for groups.  Do we know what the use case is for this?

Reply:
Don't know, but I think this was added by @djberg96 so possibly he can comment?  At least the specs were authored by him (pretty sure).

Reply:
@NickLaMuro Not that I recall. Do you have a commit? I couldn't find one in the log with my name for this file.

Reply:
@djberg96 woops, I think I mis-associated you with this commit:

https://github.com/ManageIQ/manageiq/commit/41067b998b9c664f51271061b36566d113afac33
https://github.com/ManageIQ/manageiq/pull/18279

And I think @kbrock was actually the one involved:

https://github.com/ManageIQ/manageiq/commit/1eb754cd25421936f17f8599b24e85a1c6bedce0
https://github.com/ManageIQ/manageiq/pull/5164

Reply:
Yea, we use sequence for policies, widgets, actions, and groups.

the value didn't seem to matter for groups, just so that it is unique within the user's groups or the region maybe? I think this is where the scope comes into play. And where rails will audibly complain

Also, I'd like to say that we want to remove nil values from the sequence so ordering would be easier and more consistent.

Reply:
also, this `maximum` is executing an extra sql query on create. sure would like to get rid of that

Reply:
I guess I'm less interested in how sequences work as opposed to what feature does sequences provide to the end user? (I don't understand it for polcies and actions either, unless that the order or execution in a policy run, but why?  Nor for widgets, unless that's a position on a dashboard, but then wouldn't that be defined in the dashboard?)